modules/BuildTools.Tasks/CreateSourceLink.cs

```diff
@@ -34,7 +34,9 @@ public class CreateSourceLink : Task
         public override bool Execute()
         {
             var rootDirectory = Path.Combine(Path.GetFullPath(RootDirectory), "*");
-            rootDirectory = rootDirectory.Replace("\\", "/");
+
+            rootDirectory = rootDirectory.Replace("/", "\\");
```

Comment:
If I understand [this doc](https://github.com/dotnet/core/blob/3cc74f3cd7ff6ca2d94e4e1b882e8551b5c32a50/Documentation/diagnostics/source_link.md#examples) correctly, we should normalize `/` and `\` to `System.IO.Path.DirectorySeparatorChar` so LInux builds also work.

Reply:
The paths should be the same as passed to the csc. The only escaping that should be done here is JSON escaping (```\``` -> ```\\```).

Reply:
Didn't know that existed. Thanks for the pointer.

Do you have any plans to make this a feature of the SDK? I'm sure other devs would like this too, but it took a lot of research to figure this out. Even then, we still don't have it 100% working.

Reply:
Yes, we do. Working on it.

Reply:
BTW, do you build with Deterministic=true?

Reply:
👍 cool.

We haven't changed the default AFAIK. Deterministic is true by default, right?

Reply:
Yes. However, to get fully deterministic build you'll also need to set PathMap. This will make paths embedded in the DLL and PDBs independent of the directory the repo was cloned to.
See https://github.com/dotnet/roslyn/blob/master/build/Targets/Settings.props#L136 and https://github.com/dotnet/roslyn-tools/blob/master/sdks/RepoToolset/tools/Workarounds.targets#L103

Reply:
Another thing to consider is submodules. They complicate things.

Reply:
Submodules complicate things for your SourceLink targets or for deterministic builds? How do they complicate things? Our packages are produced by Universe, which does use submodules.

Reply:
For SourceLink, because the simple logic implemented in [SourceLink.targets](https://github.com/dotnet/roslyn-tools/blob/master/sdks/RepoToolset/tools/SourceLink.targets) doesn't automatically include adding submodule entries into source link file. Once we have source link integrated in dotnet SDK we will handle them.

Reply:
Submodules isn't really a problem for us at the moment. We do not have any projects that produce DLLs from source files consumed from a submodule. We use submodules, but only for coordinating a build of [multiple repos together](https://github.com/aspnet/Universe/tree/dev/modules), not for sharing sources between them.

Reply:
Good :)